src/utils/wrap_thread.cpp

```diff
@@ -66,7 +66,6 @@ bool wrap_thread::try_join(std::chrono::duration<double> duration) {
 	} else {
```

Comment:
Create ATTR_NODISCARD - in platform.hpp
and use it on try_join
also comment

/// caller MUST check the returned value, if it is false then thread is NOT joined so it will be probably UB to e.g. use shared variables at this point



// g++, clang++                                                                                                                    
// TODO same on mac clang?                                                                                                                                                                                         
#define ATTR_NODISCARD __attribute__((warn_unused_result))                                                                         
// TODO msvc?  mingw?                                                                                                              
                                                                                                                                   
ATTR_NODISCARD bool tryme() { return 1; }                                                                                          
                                                                                                                                   
int main() {                                                                                                                       
  bool x = tryme();                                                                                                                
  return x;                                                                                                                        
}